tests/test-lxd-rbac

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+set -eu
+
+waitSnapdSeed() (
+  set +x
+  for i in $(seq 60); do # Wait up to 60s.
+    if systemctl show snapd.seeded.service --value --property SubState | grep -qx exited; then
+      return 0 # Success.
+    fi
+
+    sleep 1
+  done
+
+  echo "snapd not seeded after ${i}s"
+  return 1 # Failed.
+)
+
+cleanup() {
+    echo ""
+    if [ "${FAIL}" = "1" ]; then
+        echo "Test failed"
+        exit 1
+    fi
+
+    echo "Test passed"
+    exit 0
+}
+
+FAIL=1
+trap cleanup EXIT HUP INT TERM
+
+# Wait for snapd seeding
+waitSnapdSeed
+
+# Install LXD
+while [ -e /usr/bin/lxd ]; do
+    apt-get remove --purge --yes lxd lxd-client lxcfs liblxc1
+done
+apt-get remove --purge cloud-init --yes
+snap remove lxd || true
+snap install lxd --channel=latest/edge
+apt-get install --no-install-recommends --yes jq
+snap install bhttp
+lxd waitready --timeout=300
+
+# Configure LXD
+lxc storage create default zfs
+lxc profile device add default root disk path=/ pool=default
+lxc network create lxdbr0
+lxc profile device add default eth0 nic network=lxdbr0 name=eth0
+
+# Configure RBAC
+curl https://canonical-lxd.monstermunchkin.dev/config/rbac-test.service -o rbac-test.service
+lxc config set rbac.agent.private_key "$(jq -r .key.private < rbac-test.service)"
+lxc config set rbac.agent.public_key "$(jq -r .key.public < rbac-test.service)"
+lxc config set rbac.agent.url "$(jq -r .agents[0].url < rbac-test.service)"
+lxc config set rbac.agent.username "$(jq -r .agents[0].username < rbac-test.service)"
+lxc config set rbac.api.key 4BG3ukH0XFbskFH15Se11C4K0ENBiO4fxc917R0yJRo
+lxc config set rbac.api.url https://services.monstermunchkin.dev/rbac
+lxc config set core.https_address :8443
+
+# Setup remote
+(
+cat << EOF
+default-remote: local
+remotes:
+  images:
+    addr: https://images.linuxcontainers.org
+    protocol: simplestreams
+    public: true
+  local:
+    addr: unix://
+    public: false
+  rbac-user:
+    addr: https://127.0.0.1:8443
+    protocol: lxd
+    auth_type: candid
+    protocol: lxd
+    public: false
+  rbac-admin:
+    addr: https://127.0.0.1:8443
+    protocol: lxd
+    auth_type: candid
+    protocol: lxd
+    public: false
+aliases: {}
+EOF
+) > ~/snap/lxd/common/config/config.yml
```

Comment:
I know it's a copy-n-pasted bit but would you mind loosing the subshell and use `cat << EOF > ~/snap/lxd/common/config/config.yml` instead?